src/swap/components/SwapSettings.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useCallback, useState, useRef, useEffect } from 'react';
+import { Children, cloneElement, isValidElement, useMemo } from 'react';
+import { cn } from '../../styles/theme';
+import { useIcon } from '../../wallet/hooks/useIcon';
+import type { SwapSettingsReact } from '../types';
+
+export function SwapSettings({
+  children,
+  className,
+  icon = 'swapSettings',
+  text = '',
+}: SwapSettingsReact) {
+  const [isOpen, setIsOpen] = useState(false);
+  const dropdownRef = useRef<HTMLDivElement>(null);
+  
+  const childrenArray = useMemo(() => {
+    return Children.toArray(children).map((child) => {
+      if (isValidElement(child)) {
+        // @ts-ignore
+        return cloneElement(child);
+      }
+      return child;
+    });
+  }, [children]);
+
+  const handleToggle = useCallback(() => {
+    setIsOpen(!isOpen);
+  }, [isOpen]);
+
+  const handleClickOutsideComponent = useCallback((event: MouseEvent) => {
+    if (dropdownRef.current && !dropdownRef.current.contains(event.target as Node)) {
+      setIsOpen(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutsideComponent);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutsideComponent);
+    };
+  }, [handleClickOutsideComponent]);
+
+  const iconSvg = useIcon({ icon });
+
+  return (
+    <div
+      className={cn(
+        className,
+        'flex w-full items-center justify-end space-x-1',
```

Comment:
hmm i believe the order here is important. can you swap line 48 and 49 so that classname isn't overridden?

Reply:
Nice catch!